File: app/src/main/java/org/zwsmith/myapplication/presentation/MyListAdapter.java
package org.zwsmith.myapplication.presentation;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import org.zwsmith.myapplication.R;

import java.util.ArrayList;

public class MyListAdapter extends RecyclerView.Adapter<MyListAdapter.ViewHolder> {
    private ArrayList<String> titles;

    MyListAdapter(ArrayList<String> titles) {
        this.titles = titles;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.rv_item_layout, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        TextView titleTextView = holder.itemView.findViewById(R.id.title_tv);
        titleTextView.setText(titles.get(position));
    }

    @Override
    public int getItemCount() {
        return titles.size();
    }

    public void updateTitles(ArrayList<String>  titles) {
        this.titles = titles;
        notifyDataSetChanged();
    }

    static class ViewHolder extends RecyclerView.ViewHolder {
        View itemView;

        ViewHolder(View view) {
            super(view);
            itemView = view;
        }
    }
}

File: app/src/main/java/org/zwsmith/myapplication/presentation/MyListAdapter.kt
package org.zwsmith.myapplication.presentation

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import org.zwsmith.myapplication.R

import java.util.ArrayList

class MyListAdapter internal constructor(private var titles: ArrayList<String>?) : RecyclerView.Adapter<MyListAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.rv_item_layout, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val titleTextView = holder.itemView.findViewById(R.id.title_tv)
        titleTextView.setText(titles!![position])
    }

    override fun getItemCount(): Int {
        return titles!!.size
    }

    fun updateTitles(titles: ArrayList<String>) {
        this.titles = titles
        notifyDataSetChanged()
    }

    internal class ViewHolder(var itemView: View) : RecyclerView.ViewHolder(itemView)
}
